Admin/Extension/LockExtension.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin\Extension;
+
+use Sonata\AdminBundle\Admin\AdminExtension;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Model\LockInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Emmanuel Vella <vella.emmanuel@gmail.com>
+ */
+class LockExtension extends AdminExtension
+{
+    protected $lockVersions;
+
+    public function __construct()
+    {
+        $this->lockVersions = new \SplObjectStorage();
```

Comment:
As it's a service, it was to avoid potential bugs. But I'm not sure it's useful. To you prefer a simple $version variable ?